khmer/kfile.py

```diff
@@ -46,7 +48,21 @@ def check_input_files(file_path, force):
         if not force:
             sys.exit(1)
     else:
-        if os.stat(file_path).st_size == 0:
+        is_empty = False
+        file_size = os.stat(file_path).st_size
+
+        if file_size == 0:
+            is_empty = True
+        elif file_path.endswith('gz'):
+            fh = gzip.open(file_path, 'rb')
```

Comment:
See https://github.com/dib-lab/screed/blob/master/screed/openscreed.py#L40 for what screed does. Perhaps this functionality could move there via an optional argument.